### ISSUE TITLE
Figure out a group calendar that also fits individuals

### ISSUE THREAD

**Issue by prakesome** [pyatl/shovel#8]

**Comment by prakesome** [comment]
assigning to @dcsim0n

**Comment by dcsim0n** [comment]
Hello, test 123

**Comment by dcsim0n** [comment]
- All organizational events can be public, even if the meeting invites are not accessible to public
- The shared calendar will have all pyAtl events and be visible to anyone interested in pyAtl
- Discord calendar can sync a google calendar with discord
- Standard wordpress calendar is free: https://wordpress.org/plugins/ics-calendar/

**Comment by prakesome** [comment]
@dcsim0n there's also https://wordpress.org/plugins/the-events-calendar/ which mimics meetup.

**Comment by dcsim0n** [comment]
This is a list of three popular WordPress plugins that I researched for using as our main calendar for pyAtl.

Based on my research and some limited testing with my own wordpress and discord servers, I think option #3, the Event Calendar, is the best fit for our situation right now and gives us a little room for growth. The interface is simple, and provides basic features that will be easy to use and help community members stay informed.

The basic requirements for a calendar were:

1. All organizational events should be public, even if the meeting venue (discord / private venue, etc) is not open to the public.
2. The shared calendar will have all pyAtl events and be visible to anyone interested in pyAtl (pyAtl.dev/events)
3. Be able to post notifications to discord.

The plugins I researched are:
1. [ICS Calendar](https://wordpress.org/plugins/search/ics+calendar/): import events from an ics calendar url (google calendar) and dynamically create a wordpress calendar visible on the public pyatl.dev site.
  - Pros: uses familiar calendar interface to manage public events (google calendar)
  - Cons: the ‘source of truth’ calendar is owned by an individual’s google account, so changing ownership might be cumbersome.
  - There are a variety of bots that can sync with a google calendar too, so both wordpress and discord could point at the same calendar. Some bots even send reminders and information about updates.

2. [Timely](https://wordpress.org/plugins/all-in-one-event-calendar/): feature right event calendar with advanced ticketing and promotion features

  - Pros: lots of advanced features for promoting events and handling rsvp/ticket sales
  - Cons: requires time.ly account (3rd party) and the free tier has a limit of 100 events / year.
  - It’s not clear how timely would integrate with discord. I didn’t find any ready made plugins, custom wordpress hook would probably be necessary.

3. [Event Calendar](https://wordpress.org/plugins/the-events-calendar/): create events using the built in WordPress block editor. 

  - Pros: access can be managed by wordpress and all event data is hosted by pyatl.dev, simple interface, allows sharing to google calendar.
  - Cons: basic features that we may outgrow with time. Importing events from google calendar or meetup is a paid service. 
  - WordPress and Discord hooks can be used to push directly to discord when new events are created. Some plugins exist, but features are limited: https://github.com/RealHartlMax/Calendar-to-Discord

**Comment by prakesome** [comment]
Great, thank you!Will go ahead with the events plugin. Thanks for the work, Dana. On Dec 5, 2023, at 6:06 PM, Dana Simmons ***@***.***> wrote:﻿
This is a list of three popular WordPress plugins that I researched for using as our main calendar for pyAtl.
Based on my research and some limited testing with my own wordpress and discord servers, I think option #3, the Event Calendar, is the best fit for our situation right now and gives us a little room for growth. The interface is simple, and provides basic features that will be easy to use and help community members stay informed.
The basic requirements for a calendar were:
All organizational events should be public, even if the meeting venue (discord / private venue, etc) is not open to the public.
The shared calendar will have all pyAtl events and be visible to anyone interested in pyAtl (pyAtl.dev/events)
Be able to post notifications to discord.
The plugins I researched are:
ICS Calendar: import events from an ics calendar url (google calendar) and dynamically create a wordpress calendar visible on the public pyatl.dev site.
Pros: uses familiar calendar interface to manage public events (google calendar)
Cons: the ‘source of truth’ calendar is owned by an individual’s google account, so changing ownership might be cumbersome.
There are a variety of bots that can sync with a google calendar too, so both wordpress and discord could point at the same calendar. Some bots even send reminders and information about updates.
Timely: feature right event calendar with advanced ticketing and promotion features
Pros: lots of advanced features for promoting events and handling rsvp/ticket sales
Cons: requires time.ly account (3rd party) and the free tier has a limit of 100 events / year.
It’s not clear how timely would integrate with discord. I didn’t find any ready made plugins, custom wordpress hook would probably be necessary.
Event Calendar: create events using the built in WordPress block editor.
Pros: access can be managed by wordpress and all event data is hosted by pyatl.dev, simple interface, allows sharing to google calendar.
Cons: basic features that we may outgrow with time. Importing events from google calendar or meetup is a paid service.
WordPress and Discord hooks can be used to push directly to discord when new events are created. Some plugins exist, but features are limited: https://github.com/RealHartlMax/Calendar-to-Discord

**Comment by dcsim0n** [comment]
Sounds good. Since this task was mostly research I'm going to move this to done and open some new issues for implementing.

**Comment by prakesome** [comment]
Yes perfect. On Dec 6, 2023, at 11:13 AM, Dana Simmons ***@***.***> wrote:﻿
Sounds good. Since this task was mostly research I'm going to move this to done and open some new issues for implementing.
